src/kixi/plugsocket.clj

```diff
@@ -0,0 +1,38 @@
+(ns kixi.plugsocket
+  (:import [java.io OutputStream FileOutputStream]
+           org.apache.poi.xslf.usermodel.XMLSlideShow
+           ))
+
+(defmacro assert-type [value expected-type]
+  `(when-not (isa? (class ~value) ~expected-type)
+     (throw (IllegalArgumentException.
+             (format "%s is invalid. Expected %s. Actual type %s, value: %s"
+                     (str '~value) ~expected-type (class ~value) ~value)))))
+
+(defn save-powerpoint-into-stream!
+  "Save the workbook into a stream.
+  The caller is required to close the stream after saving is completed."
+  [^OutputStream stream ^XMLSlideShow powerpoint]
+  (assert-type powerpoint XMLSlideShow)
+  (.write powerpoint stream))
+
+(defn save-powerpoint-into-file!
+  "Save the workbook into a file."
+  [^String filename ^XMLSlideShow powerpoint]
+  (assert-type powerpoint XMLSlideShow)
```

Comment:
is `assert-type` needed for these given that the multimethod is switching on the class?

Reply:
That was torn from kixi.large.legacy, so you'll have to tell me?

Reply:
kixi.large.legacy is just copying over the stuff from docjure w/o changing anything that wasn't needed to update the underlying poi libs.

Reply:
Will remove in following PR